lib/common_test/src/cth_log_redirect.erl

```diff
@@ -169,10 +175,11 @@ log(#{msg:={report,Msg},meta:=#{domain:=[otp,sasl]}}=Log,Config) ->
                     do_log(add_log_category(Log,sasl),Config)
             end
     end;
+%% Prevent recursively logging messages we have seen
+log(#{meta:=Meta}=_Log, _Config) when is_map_key(?MODULE, Meta) ->
+    ok;
 log(#{meta:=#{domain:=[otp]}}=Log,Config) ->
     do_log(add_log_category(Log,error_logger),Config);
-log(#{meta:=#{domain:=_}},_) ->
-    ok;
```

Comment:
can you give more details? what was the unwanted flow?

Reply:
From what I gathered when writing this, the previous behaviour would ignore redirect for some messages. I figured that the log redirect should probably capture _all_ log messages, so removed it. For example the HTML logs for `httpd_SUITE:bad_hex/1` with this change include:
```erl
*** System report during httpd_SUITE:bad_hex/1 in http_1_0 2023-06-12 23:45:06.509 ***

=ERROR REPORT==== 12-Jun-2023::23:45:06.509079 ===
   Server: httpd_test
 Protocol: HTTP
Transport: TLS
     Host: 127.0.1.1:40797
     Peer: 127.0.0.1:50042
   Reason: [{statuscode,400},
            {description,"Bad Request"},
            {details,{malformed_syntax,"http://www.erlang.org/%skalle"}}]
```
Without this change, this report is not included.

Reply:
Also, since we're already on the review comments in here: I am not sure if the "prevent recursively logging" does what I intend it to. If we added the key I don't think we're passed the log message again because it's immutable. I did see some test cases that looked suspiciously like they were recursively logging themselves again but I'm not sure it was caused by this one...